packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -251,6 +251,52 @@ describe('CDK Include', () => {
     );
   });
 
+  test("correctly parses templates with parameters", () => {
+    const cfnTemplate = includeTestTemplate(stack, 'bucket-with-parameters.json');
+    const param = cfnTemplate.getParameter('BucketName');
+    const bucket = new s3.CfnBucket(stack, "newBucket", {"bucketName": param.valueAsString});
+
+    expect(stack).toMatchTemplate(
```

Comment:
BTW. For this assertion, you can use the splice operator to make it more dry. See here: https://github.com/aws/aws-cdk/pull/8529/files#diff-c2f3aef0d1898e518e05ca13579250e8R284 for an example. So this would be:

```ts
      const originalTemplate = loadTestFileToJsObject('bucket-with-parameters.json');
      expect(stack).toMatchTemplate({
        "Resources": {
           ...originalTemplate.Resources,
          "newBucket": {
            "Type": "AWS::S3::Bucket",
            "Properties": {
              "BucketName": {
                "Ref": "BucketName",
              },
            },
         },
         "Parameters": {
           ..originalTemplate.Parameters,
        },
      });